src/box/xrow.c

```diff
@@ -2386,7 +2422,18 @@ xrow_decode_fetch_snapshot(const struct xrow_header *row,
 	memset(req, 0, sizeof(*req));
 	struct replication_request base_req = {
 		.version_id = &req->version_id,
+		.is_checkpoint_join = &req->is_checkpoint_join,
+		.checkpoint_vclock = &req->checkpoint_vclock,
+		.checkpoint_lsn = &req->checkpoint_lsn,
 	};
+	/*
+	 * IS_CHECKPOINT_JOIN, CHECKPOINT_VCLOCK and CHECKPOINT_LSN are optional
+	 * keys, they are used exclusively by CDC now, so we need to set
+	 * default values for vclock and lsn before decoding.
+	 */
+	req->is_checkpoint_join = false;
+	vclock_clear(&req->checkpoint_vclock);
+	req->checkpoint_lsn = 0;
```

Comment:
There is `memset(0)` above which does all the same. These lines don't seem necessary.

Reply:
`vclock_clear` does a lot more:

https://github.com/tarantool/tarantool/blob/7db4de75f8bd975066635f942ee49f33a2209353/src/lib/vclock/vclock.h#L128-L134

This -1 signature is what I need. `memset` won't do that for me. And I strongly rely on it in `memtx_engine_prepare_checkpoint_join`

So, let's leave everything as it is.

Reply:
But you surely don't need `is_checkpoint_join = false;` and `checkpoint_lsn = 0;`?

Reply:
I don't) Fixed